lib/Remote/DesiredCapabilities.php

```diff
@@ -163,25 +163,10 @@ public function setJavascriptEnabled($enabled)
     }
 
     /**
-     * @todo Remove side-effects - not change eg. ChromeOptions::CAPABILITY from instance of ChromeOptions to an array
      * @return array
      */
     public function toArray()
     {
-        if (isset($this->capabilities[ChromeOptions::CAPABILITY]) &&
```

Comment:
the problem with these changes that it's a B/C break, it's changing the shape of a return, before this PR "ChromeOptions::CAPABILITY" and "FirefoxDriver::PROFILE" are converted into an array but after this PR they will be returned as is.

Reply:
Arguably this is correct and the fact that they return objects is a bug, but I agree that this would be a change in behaviour and necessity a major version bump.